src/lib/pubkey/dilithium/dilithium_common/dilithium_symmetric_primitives.h

```diff
@@ -0,0 +1,323 @@
+/*
+* Asymmetric primitives for dilithium
+* (C) 2022 Jack Lloyd
+* (C) 2022 Manuel Glaser, Michael Boric, René Meusel - Rohde & Schwarz Cybersecurity
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#ifndef BOTAN_DILITHIUM_ASYM_PRIMITIVES_H_
+#define BOTAN_DILITHIUM_ASYM_PRIMITIVES_H_
+
+#include <botan/dilithium.h>
+#include <botan/stream_cipher.h>
+
+#include <botan/internal/shake.h>
+
+#include <memory>
+#include <vector>
+
+namespace Botan {
+
+/**
+* Adapter class that uses polymorphy to distinguish
+* Dilithium "common" from Dilithium "AES" modes.
+*/
+class Dilithium_Symmetric_Primitives
+   {
+   public:
+      enum class XofType
+         {
+         k128,
+         k256
+         };
+
+   public:
```

Comment:
That's an intentional habit of mine: using visibility declarations to visually structure class definitions. Such as:

```C++
class Foo
   {
   public:
      // internal types

   public:
      // public methods

   private:
      // private methods

   private:
      // private members
   };
```

Should I stop doing that?

Reply:
TBH I would prefer literal comments along the lines you included in your example, rather than the redundant specifiers which do not have immediate meaning to a reader. But it's not a big deal.